internal/table/client.go

```diff
@@ -348,59 +356,60 @@ func (c *Client) BulkUpsert(
 // chunkBulkUpsertRequest splits a BulkUpsertRequest into smaller chunks if it exceeds the maximum message size.
 // It recursively divides the request's rows into smaller requests while preserving the original request structure.
 // If the request is smaller than the maximum size or contains fewer than two rows, it returns the original request.
-func chunkBulkUpsertRequest(req *Ydb_Table.BulkUpsertRequest, maxBytes int) ([]*Ydb_Table.BulkUpsertRequest, error) {
-	// not a row bulk upsert request -> ret original request
-	if req.Rows == nil || req.Rows.Value == nil {
-		return []*Ydb_Table.BulkUpsertRequest{req}, nil
-	}
-
+func chunkBulkUpsertRequest(
+	dst *[]*Ydb_Table.BulkUpsertRequest,
```

Comment:
Pointer to slice is uncommon pattern, what about return slice with appended requests, similar to builtin `append`?